packages/x-date-pickers/src/internals/models/fields.ts

```diff
@@ -18,4 +18,5 @@ export interface BaseFieldProps<
   format?: string;
   disabled?: boolean;
   ref?: React.Ref<HTMLDivElement>;
+  ownerState?: any;
```

Comment:
Since the field is used as a slot inside the picker, it receives an `ownerState`

And some (most?) `@mui/material` components don't like when you pass an `ownerState` to them (it overrides their own `ownerState`).
So I had to add it to the typing here to be able to not forward it to the `Button` in the updated demos

Reply:
Do we want to update the type of it to `PickersOwnerState`? 🤔

Reply:
We need #14994